eos-reclaim-swap.service

```diff
@@ -0,0 +1,14 @@
+[Unit]
+Description=Remove swap partitions and reclaim their space
+Before=multi-user.target systemd-update-done.service
+After=swap.target
+After=eos-enable-zram.service # This relies on eos-enable-zram.service being Type=oneshot
```

Comment:
Maybe changing eos-enable-zram.service to have `DefaultDependencies=no` and `Requires=swap.target` (it already has `After=swap.target` but I believe that is essentially a noop with `DefaultDependencies=yes`). Or even trying to order eos-reclaim-swap.service `Before=swap.target` and `After=` some essential services it needs like udevd, although I'm not sure if that would work.

I'll experiment with the ideas above and if it does not help, I'll go for your suggestion. I would rather avoid dropping another random file in `/var`.

Reply:
Actually, any service ordered `Before=systemd-update-done.service` needs `DefaultDependencies=no` since systemd-update-done.service is ordered before sysinit.target. So making eos-reclaim-swap.service `DefaultDependencies=no` and ordering it `Before=swap.target` is the way to go here and it also makes it completely independent from eos-enable-zram.service.

Reply:
Trying to move this service and its dependencies to early boot to be able to use `ConditionsNeedsUpdate` started creating a few unexpected cycles, so I decided to keep things simple and give this service its own state file.